samza-core/src/main/java/org/apache/samza/zk/ZkCoordinationServiceFactory.java

```diff
@@ -33,15 +33,11 @@
 public class ZkCoordinationServiceFactory implements CoordinationServiceFactory {
   private final static Logger LOG = LoggerFactory.getLogger(ZkCoordinationServiceFactory.class);
 
-  // TODO - Why should this method be synchronized?
   synchronized public CoordinationUtils getCoordinationService(String groupId, String participantId, Config config) {
```

Comment:
It used to be called from multiple places, but now we call it only from one place. So I don't think we need to sync it.